src/Tasks/Microsoft.NET.Build.Tasks/NuGetPackageResolver.cs

```diff
@@ -31,6 +31,19 @@ public string GetPackageDirectory(string packageId, NuGetVersion version)
             return _packagePathResolver.GetPackageDirectory(packageId, version);
         }
 
+        public string GetPackageDirectory(string packageId, NuGetVersion version, out string packageRoot)
+        {
+            string packageDir = null;
+            packageRoot = null;
+            var pkginfo = _packagePathResolver.GetPackageInfo(packageId,version);
+            if (pkginfo != null)
+            {
+                packageDir  = pkginfo.PathResolver.GetInstallPath(packageId, version);
+                packageRoot = pkginfo.PathResolver.GetVersionListPath("");  //TODO Check with nuget  if this is the correct way to get the package root
+            }
+            return packageDir;
```

Comment:
Shouldnt the invariant of this function be that it will always return non-null value?